Update 2020-02-29-f12.markdown (#2)
kalehmann/blog-kalehmann-de: _posts/2020-02-29-f12.markdown

  - Coding
---

I've already had to create bootable floppy images in some of my projects.
This process has always been a hassle, especially automating it.
So I've decided to write my own all-in-one solution to create FAT12 formatted
images - f12.
f12 is a program written in C, that provides various functions to handle FAT12
formatted images, for example moving files to the image, dumping files from the
image and listing contents of directories on the image.
But the most import feature of all for myself is creating new images with a set

> file and installing a bootloader in one command, but let's start at the

Reviewer: ```suggestion
of files and installing a bootloader in one command, but let's start at the
```

Reviewer: Hier rede ich von mehreren Dateien, die ich gerne auf das Image schreiben würde.

Author: Gut dann: But the most important (Grad noch ne fehler gesehen lul) feature of all for myself, is creating new images with set files and installing...

Reviewer: Stimmt, `important` habe ich auch übersehen

> file and installing a bootloader in one command, but let's start at the

Reviewer: ```suggestion
But the most important feature of all for myself is creating new images with a set
of files and installing a bootloader in one command, but let's start at the
```

{% highlight shell %}
# Create an empty FAT12 formatted image
mkfs.msdos -C test.flp 1440

> # Write a bootloader for it

Reviewer: ```suggestion
# Write a bootloader to it
```

Hier möchte ich sagen, dass ich den Bootlaoder **auf** das Image schreiben will

Author: Gut, dann: Write the bootloader for it

mkfs.msdos -C test.flp 1440
# Write a bootloader to it
dd if=bootloader.bin of=test.flp conv=notrunc bs=512 count=1

> # Do some hacking to configure the bootloader

Reviewer: ```suggestion
# Do some hack to configure the bootloader
```

These commands have some disadvantages.
At first they require superuser privileges to mount the images.
This should not be necessary for such a simple process and may not work in

> containerised  environments.

Reviewer: ```suggestion
containerized  environments.
```

Ich glaube, das war schon vorher richtig so

* https://hackernoon.com/what-is-containerization-83ae53a709a6
* https://www.ibm.com/cloud/learn/containerization

Author: Gut was neues dazu gelernt

mkfs.msdos -C test.flp 1440
# Write a bootloader to it
dd if=bootloader.bin of=test.flp conv=notrunc bs=512 count=1

> # Do some hacking to configure the bootloader

Reviewer: ```suggestion
# Do some hack to configure the bootloader
```
